src/main/java/control/TimelineController.java

```diff
@@ -93,4 +102,74 @@ private void initializeCameraTimelines() {
             scriptingProject.addCameraTimeline(timelineN);
         }
     }
+    
+    /**
+     * Save the current project state to file.
+     * A file chooser window will be opened to select a file
+     */
+    public void save() {
```

Comment:
I made a save and load (or read and write but you can rename them) function in the ScriptingProject class. Could you move this code to that location since I believe saving and loading belongs to a project and not a timelinecontroller